### ISSUE TITLE
[associative] v6.0.0 error on import

### ISSUE THREAD

**Issue by fredkilbourn** [thi-ng/umbrella#321]
- @thi.ng/associative v6.0.0
- nodejs v16.11.0
- npm v8.0.0

```javascript
const associative = await import("@thi.ng/associative");
console.log( associative );
```

```console
node test2.js
node:internal/errors:464
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/vbot/srv/vbot/node_modules/@thi.ng/associative/api' imported from /home/vbot/srv/vbot/node_modules/@thi.ng/associative/index.js
    at new NodeError (node:internal/errors:371:5)
    at finalizeResolution (node:internal/modules/esm/resolve:416:11)
    at moduleResolve (node:internal/modules/esm/resolve:932:10)
    at Loader.defaultResolve [as _resolve] (node:internal/modules/esm/resolve:1044:11)
    at Loader.resolve (node:internal/modules/esm/loader:89:40)
    at Loader.getModuleJob (node:internal/modules/esm/loader:242:28)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:40)
    at link (node:internal/modules/esm/module_job:75:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

**Comment by postspectacular** [comment]
@fredkilbourn I did some more reading and realized where the mistake is.

- https://blog.logrocket.com/es-modules-in-node-today/
- https://devblogs.microsoft.com/typescript/announcing-typescript-4-5-beta/

Currently was still using package-internal imports like this:

```ts
import { __dispatch } from "./internal/dispatch";
```

However, these kind of imports are now seemingly deprecated (and only still supported via the above mentioned `--experimental-specifier-resolution=node` CLI flag). So I just updated all such imports to add `.js` suffixes:

```ts
import { __dispatch } from "./internal/dispatch.js";
```

...and now we should be able to use `node` without any other flags!

([see commits](https://github.com/thi-ng/umbrella/compare/467a148...develop)...)

New releases later today...
